tests/core/middleware/test_eth_tester_middleware.py

```diff
@@ -20,3 +27,157 @@ def test_get_block_formatters(w3):
     keys_diff = all_block_keys.difference(latest_block_keys)
     assert len(keys_diff) == 1
     assert keys_diff.pop() == "mixHash"  # mixHash is not implemented in eth-tester
+
+
+sample_address_1 = "0x0000000000000000000000000000000000000001"
+sample_address_2 = "0x0000000000000000000000000000000000000002"
+
+
+@pytest.mark.parametrize(
+    "w3_accounts, w3_coinbase, method, from_field_added, from_field_value",
+    (
+        (sample_address_1, sample_address_2, "eth_call", True, sample_address_2),
+        (
+            sample_address_1,
+            sample_address_2,
+            "eth_estimateGas",
+            True,
+            sample_address_2,
+        ),
+        (
+            sample_address_1,
+            sample_address_2,
+            "eth_sendTransaction",
+            True,
+            sample_address_2,
+        ),
+        (sample_address_1, sample_address_2, "eth_gasPrice", False, None),
+        (sample_address_1, sample_address_2, "eth_blockNumber", False, None),
+        (sample_address_1, sample_address_2, "meow", False, None),
+        (sample_address_1, None, "eth_call", True, sample_address_1),
+        (sample_address_1, None, "eth_estimateGas", True, sample_address_1),
+        (sample_address_1, None, "eth_sendTransaction", True, sample_address_1),
+        (sample_address_1, None, "eth_gasPrice", False, None),
+        (sample_address_1, None, "eth_blockNumber", False, None),
+        (sample_address_1, None, "meow", False, None),
+        (None, sample_address_2, "eth_call", True, sample_address_2),
+        (None, sample_address_2, "eth_estimateGas", True, sample_address_2),
+        (None, sample_address_2, "eth_sendTransaction", True, sample_address_2),
+        (None, sample_address_2, "eth_gasPrice", False, sample_address_2),
+        (None, sample_address_2, "eth_blockNumber", False, sample_address_2),
+        (None, sample_address_2, "meow", False, sample_address_2),
+        (None, None, "eth_call", True, None),
+        (None, None, "eth_estimateGas", True, None),
+        (None, None, "eth_sendTransaction", True, None),
+        (None, None, "eth_gasPrice", False, None),
+        (None, None, "eth_blockNumber", False, None),
+        (None, None, "meow", False, None),
+    ),
+)
+def test_default_transaction_fields_middleware(
+    w3_accounts, w3_coinbase, method, from_field_added, from_field_value
+):
+    w3_accounts = [w3_accounts]
```

Comment:
Instead of re-defining the variable here, I think we should either make the mock return the list with a variable more appropriately named `w3_account` (singular):

```py
mock_w3.eth.accounts = [w3_account]  # and the variable is renamed to w3_account (singular)

# async:
async def mock_async_accounts():
    return [w3_account]  # singular
```

or... we can just pass in a list of accounts into the tests themselves and keep the variable as `w3_accounts` (plural). I vote this option and we can change one of the tests to return more than one account in the list to make sure that works as intended as well... I believe it always returns `accounts[0]`.

Reply:
Makes sense to have the mock return what the function normally would. I've updated the sample address constants accordingly. It will still only ever choose index 0 of the list of addresses though, as that's how the `guess_from` function is written.